File: leetcode2/1easy/노형민/Q3633.cs
public class Solution
{
    public int EarliestFinishTime(int[] landStartTime, int[] landDuration, int[] waterStartTime, int[] waterDuration)
    {
        int minTime = 10000;
        for (int i = 0; i < landStartTime.Length; i++)
        {
            for (int j = 0; j < waterStartTime.Length; j++)
            {
                int currentTime = landStartTime[i] + landDuration[i];
                if (waterStartTime[j] <= currentTime)
                {
                    currentTime += waterDuration[j];
                }
                else
                {
                    currentTime = waterStartTime[j] + waterDuration[j];
                }

                if (currentTime < minTime)
                {
                    minTime = currentTime;
                }
            }
        }

        for (int i = 0; i < waterStartTime.Length; i++)
        {
            for (int j = 0; j < landStartTime.Length; j++)
            {
                int currentTime = waterStartTime[i] + waterDuration[i];
                if (landStartTime[j] <= currentTime)
                {
                    currentTime += landDuration[j];
                }
                else
                {
                    currentTime = landStartTime[j] + landDuration[j];
                }

                if (currentTime < minTime)
                {
                    minTime = currentTime;
                }
            }
        }

        return minTime;
    }
}

File: leetcode2/1easy/최원준/Q3633.java
package Leetcode.최원준;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

public class Q3633 {
    class Solution {
        public int earliestFinishTime(int[] landStartTime, int[] landDuration, int[] waterStartTime, int[] waterDuration) {
            int ans = Integer.MAX_VALUE;
            int n = landStartTime.length, m = waterStartTime.length;

            for (int i=0; i<n; i++) {
                int landStart = landStartTime[i], landDur = landDuration[i];
                for (int j=0; j<m; j++) {
                    int waterStart = waterStartTime[j], waterDur = waterDuration[j];
                    ans = Math.min(ans, Math.max(waterStart, landStart + landDur) + waterDur);
                    ans = Math.min(ans, Math.max(landStart, waterStart + waterDur) + landDur);
                }
            }
            return ans;
        }
    }
}
